src/dune_rules/module_compilation.ml

```diff
@@ -324,6 +325,7 @@ let build_cm
             ; A "-c"
             ; Command.Ml_kind.flag ml_kind
             ; Dep src
+            ; Hidden_deps (Dep.Set.of_files (Option.to_list original))
```

Comment:
Would be nice to leave a comment why we need the source file here. It is in fact only needed for sandboxed builds as far as I can tell.